glean-core/ffi/src/labeled.rs

```diff
@@ -0,0 +1,93 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use lazy_static::lazy_static;
+use std::convert::TryFrom;
+
+use glean_core::{metrics::*, CommonMetricData, Lifetime};
+
+use super::handlemap_ext::HandleMapExtension;
+use super::*;
+
+macro_rules! impl_labeled_metric {
```

Comment:
HA! And you said you didn't want to use macros :P That's a nice usage.

Reply:
I ... didn't ... and then I had to repeat the same 10 lines 3 times...